lib/libzrepl/mgmt_conn.c

```diff
@@ -598,6 +598,64 @@ finish_async_tasks(void)
 	return (rc);
 }
 
+/*
+ * Take a snapshot and update snapshot IO to ZAP.
+ */
+int
+uzfs_zvol_create_snapshot_update_zap(zvol_info_t *zinfo,
+    char *snap, uint64_t snapshot_io)
+{
+	int ret = 0;
+
+	mutex_enter(&zvol_list_mutex);
+
+	if (uzfs_zvol_get_status(zinfo->zv) != ZVOL_STATUS_HEALTHY) {
+		mutex_exit(&zvol_list_mutex);
+		return (ret = -1);
+	}
+
+	uzfs_zvol_store_last_committed_io_no(zinfo->zv,
+	    snapshot_io -1);
+	zinfo->checkpointed_ionum = zinfo->running_ionum =
```

Comment:
can we have an assert here so that debug builds verifies that running_ionum is same as (snapshot_io - 1)?

Reply:
Let me try to add it .